src/flow_graph.py

```diff
@@ -1359,50 +1348,60 @@ def process_node(node: Node, reg_srcs: AccessRefs) -> None:
                         missing_regs.append((inp, ref))
                 inputs[inp] = sources
 
-            # Remove any clobbered accesses, or any MemoryAccesses that depend on
+            # Remove any clobbered locations, or any MemoryLocations that depend on
             # clobbered Registers. Ex: If `$v0` is clobbered, remove `$v0` and `4($v0)`
             for clob in ir.clobbers + ir.outputs:
-                for reg in reg_srcs:
-                    if access_depends_on(reg, clob) or access_may_overlap(reg, clob):
-                        reg_srcs.remove(reg)
+                for loc in loc_srcs:
+                    if location_depends_on(loc, clob) or locations_alias(loc, clob):
```

Comment:
do we still need the depends_on behavior? it should now only cover invalidating sp+offset when sp is updated, but I think we can assume that we won't have sp accesses with different values of sp within a function. (If we encounter arches/compilers that do we should still be able to resolve offsets statically in  a separate pass.)

Reply:
Good catch! 
(I agree that we can probably assume that `$sp` is consistent within a function, but it also wouldn't be hard to add a check to cover it anyways)